robots/commenter/main.go

```diff
@@ -242,29 +245,51 @@ func run(c client, query, sort string, asc, random bool, commenter func(meta) (s
 			break
 		}
 		log.Printf("Matched %s (%s)", i.HTMLURL, i.Title)
-		org, repo, number, err := parseHTMLURL(i.HTMLURL)
-		if err != nil {
-			msg := fmt.Sprintf("Failed to parse %s: %v", i.HTMLURL, err)
-			log.Print(msg)
-			problems = append(problems, msg)
-		}
-		comment, err := commenter(meta{Number: number, Org: org, Repo: repo, Issue: i})
-		if err != nil {
-			msg := fmt.Sprintf("Failed to create comment for %s/%s#%d: %v", org, repo, number, err)
-			log.Print(msg)
-			problems = append(problems, msg)
-			continue
+		if err := createComment(c, commenter, spamCeiling, i); err != nil {
+			log.Print(err.Error())
+			problems = append(problems, err.Error())
 		}
-		if err := c.CreateComment(org, repo, number, comment); err != nil {
-			msg := fmt.Sprintf("Failed to apply comment to %s/%s#%d: %v", org, repo, number, err)
-			log.Print(msg)
-			problems = append(problems, msg)
-			continue
-		}
-		log.Printf("Commented on %s", i.HTMLURL)
 	}
 	if len(problems) > 0 {
 		return fmt.Errorf("encoutered %d failures: %v", len(problems), problems)
 	}
 	return nil
 }
+
+func createComment(c client, commenter func(meta) (string, error), spamCeiling int, issue github.Issue) error {
+	org, repo, number, err := parseHTMLURL(issue.HTMLURL)
+	if err != nil {
+		return fmt.Errorf("Failed to parse %s: %v", issue.HTMLURL, err)
+	}
+	comment, err := commenter(meta{Number: number, Org: org, Repo: repo, Issue: issue})
+	if err != nil {
+		return fmt.Errorf("Failed to create comment for %s/%s#%d: %v", org, repo, number, err)
+	}
+
+	issueComments, err := c.ListIssueComments(org, repo, number)
+	if err != nil {
+		return fmt.Errorf("Failed to get comments for %s/%s#%d: %v", org, repo, number, err)
+	}
+
+	sameCommentsSequenceCounter := 0
+	for _, issueComment := range issueComments {
+		if comment != issueComment.Body {
+			sameCommentsSequenceCounter = 0
+		} else {
+			sameCommentsSequenceCounter += 1
+
+			if spamCeiling > 0 && sameCommentsSequenceCounter == spamCeiling {
```

Comment:
I don't think this logic is the right thing to do. Doing it like this would mean containing such sequence even long time ago would make the issue/PR ineligible to be included, ever?

Maybe the logic could be "x times in a recent time period"?

Reply:
Sounds fair. I'll add a new flag for the amount of time that this check would apply for.